src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -80,6 +80,33 @@ public interface ProgrammingExerciseRepository extends JpaRepository<Programming
     @EntityGraph(type = LOAD, attributePaths = "submissionPolicy")
     Optional<ProgrammingExercise> findWithSubmissionPolicyById(Long exerciseId);
 
+    List<ProgrammingExercise> findAllByProjectKey(String projectKey);
```

Comment:
project key does not have an index, we should probably add one so that this operation is fast even if there are thousands of programming exercises in the database

Reply:
added an index for the `programming_exercise_details` table: https://github.com/ls1intum/Artemis/pull/6262/commits/0cdd5327033b055c8f0d2597192c25bd5b7031c3